ext/SerializationExt.jl

```diff
@@ -0,0 +1,82 @@
+module SerializationExt
+using LRUCache
+using Serialization
+
+# Serialization of large LRUs causes a stack overflow error, so we 
+# create a custom serializer that represents LinkedNodes as Ints
+function Serialization.serialize(s::AbstractSerializer, lru::LRU{K, V}) where {K, V}
+    # Create a mapping from memory address to id
+    node_map = IdDict{LRUCache.LinkedNode{K}, Int}()
+    # Create mapping for first node
+    id = 1
+    first_node = node = lru.keyset.first
+    node_map[node] = id
+    # Go through the rest of the nodes in the cycle and create a mapping
+    node = node.next
+    while node != first_node
+        id += 1
+        node_map[node] = id
+        node = node.next
+    end
+    @assert id == length(lru) == lru.keyset.length == length(lru.dict)
+    # By this point, the first node has id 1 and the last node has id length(lru)
+    # so when deserializing, we can infer the order by the id
+    # Create the dict with ids instead of nodes
+    dict = Dict{K, Tuple{V, Int, Int}}()
+    for (key, (value, node, s)) in lru.dict
+        id = node_map[node]
+        dict[key] = (value, id, s)
+    end
```

Comment:
I would maybe change `s` here to something else, for example `sz`. `s` is also used for the `AbstractSerializer`, and while the for loop introduces a new scope, this might cause unnecessary confusion.
```suggestion
    for (key, (value, node, sz)) in lru.dict
        id = node_map[node]
        dict[key] = (value, id, sz)
    end
```